shared/protocol/src/main/java/io/pravega/shared/protocol/netty/WireCommands.java

```diff
@@ -486,7 +486,7 @@ public static Event readFrom(DataInput in, int length) throws IOException {
                 throw new InvalidMessageException("Was expecting EVENT but found: " + typeCode);
             }
             int eventLength = in.readInt();
-            if (eventLength != length - TYPE_PLUS_LENGTH_SIZE) {
+            if (eventLength > length - TYPE_PLUS_LENGTH_SIZE) {
```

Comment:
This is making the assumption that if the event length is greater than the expected length, then it is valid. That's not correct. Instead, the logic should be such that:

```
if (version == x) and (event length == expected length of version x)
then ok

if (version == y) and (event length == expected length of version y)
then ok

else throw
```

Reply:
The check here is to validate if the length of the `Event` read from the byte stream is less than `length` of the byte stream passed.  If length of `Event` is greater than the length of the byte stream then a `InvalidMessageException` will be thrown.

Reply:
This PR is about backward compatibility, so I'm assuming you need to change the verification here because the length changes with the version. The trouble is that if you make the verification less precise, you are opening up for other problematic scenarios, which is what I'm trying to avoid.

Starting from the basics, explain to me please why you need to change from `!=` to `>` in this validation? I checked both PR description and issue, and none of those explains why we need to do it.

Reply:
For example, if `eventLength` is zero, then we won't throw which is clearly incorrect.

Reply:
Yes, if the `eventLength` is read as zero due to data corruption then InvalidMessageException is not thrown (will check if there is a better solution to this issue).